src/instructlab/utils.py

```diff
@@ -215,18 +215,19 @@ def get_documents(
                 temp_dir=temp_dir,
                 skip_checkout=skip_checkout,
             )
-            file_contents = []
 
             logger.debug("Processing files...")
+            opened_files = False
             for pattern in file_patterns:
                 for file_path in glob.glob(os.path.join(repo.working_dir, pattern)):
                     if os.path.isfile(file_path) and file_path.endswith(".md"):
-                        with open(file_path, "r", encoding="utf-8") as file:
-                            file_contents.append(file.read())
+                        with open(file_path, "r", encoding="utf-8") as _file:
+                            # Success! we could open the file.
+                            # We don't actually care about the contents, though.
+                            opened_files = True
 
-            if file_contents:
-                return file_contents
-            raise SystemExit("Couldn't find knowledge documents")
+            if not opened_files:
+                raise TaxonomyReadingException("Couldn't find knowledge documents")
         except (OSError, exc.GitCommandError, FileNotFoundError) as e:
```

Comment:
(No action required) This `except` seems to do nothing at all. We'll need to clean it up later.

Reply:
yeah ... mostly tried to keep changes minimized to removing offending code that was blocking me. You're right that there's a lot of cleanup opportunity left.